src/test/mir-opt/const_allocation/32bit/rustc.main.ConstProp.after.mir

```diff
@@ -30,19 +30,19 @@ fn main() -> () {
 }
 
 alloc0 (static: FOO, size: 8, align: 4) {
-    ╾alloc17+0╼ 03 00 00 00                         │ ╾──╼....
+    ╾─a17+0x0─╼ 03 00 00 00                         │ ╾──╼....
 }
 
 alloc17 (size: 48, align: 4) {
-    0x00 │ 00 00 00 00 __ __ __ __ ╾alloc4+0─╼ 00 00 00 00 │ ....░░░░╾──╼....
-    0x10 │ 00 00 00 00 __ __ __ __ ╾alloc8+0─╼ 02 00 00 00 │ ....░░░░╾──╼....
-    0x20 │ 01 00 00 00 2a 00 00 00 ╾alloc13+0╼ 03 00 00 00 │ ....*...╾──╼....
+    0x00 │ 00 00 00 00 __ __ __ __ ╾─a4+0x0──╼ 00 00 00 00 │ ....░░░░╾──╼....
+    0x10 │ 00 00 00 00 __ __ __ __ ╾─a8+0x0──╼ 02 00 00 00 │ ....░░░░╾──╼....
+    0x20 │ 01 00 00 00 2a 00 00 00 ╾─a13+0x0─╼ 03 00 00 00 │ ....*...╾──╼....
```

Comment:
Hmpf, for the common case (offset is `0`, or otherwise small), this seems unnecessarily noisy compared to decimal (due to the prefix).

I know @RalfJung doesn't like it, but my preferred approach of printing arbitrary integers involves using a decimal digit when unambiguous (i.e. `<= 9`), and `+0` feels unnecessary so I would skip it too.

But maybe this won't matter long-term, if we improve our `ty::Const` printing to move beyond the need for these (otherwise really cool!) allocation dumps. (We would need to detect if the allocations are entirely reproducible from the pretty value, and don't have e.g. initialized padding, to avoid losing information)

Reply:
I'm okay introducing a special case of 0, to not print the offset at all.

But `alloc13+4` and elsewhere `alloc13+0x10` seems oddly inconsistent to me.